helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -121,14 +121,16 @@ public BestPossibleStateCalcStage(WagedRebalancer wagedRebalancer) {
   }
 
   private WagedRebalancer getWagedRebalancer(HelixManager helixManager,
-      Map<ClusterConfig.GlobalRebalancePreferenceKey, Integer> preferences) {
+      Map<ClusterConfig.GlobalRebalancePreferenceKey, Integer> preferences,
+      boolean isAsyncGlobalRebalance) {
     // Create WagedRebalancer instance if it hasn't been already initialized
     if (_wagedRebalancer == null) {
-      _wagedRebalancer = new WagedRebalancer(helixManager, preferences);
+      _wagedRebalancer = new WagedRebalancer(helixManager, preferences, isAsyncGlobalRebalance);
     } else {
-      // Since the preference can be updated at runtime, try to update the algorithm preference
-      // before returning the rebalancer.
-      _wagedRebalancer.updatePreference(preferences);
+      // Since the rebalance configure can be updated at runtime, try to update the rebalancer
```

Comment:
Question: Do we explicitly provide an API to update the rebalancer configuration? Or do users just update the ClusterConfig? I am hoping that this would make it into the user manual...

Reply:
The user just updates in the cluster config. Please consider this method as internal usage only.